### PR TITLE
Derive standard traits for structs

### PR DESCRIPTION
**Pull request by kennykerr**
Continuing on from #3015, this further derives the standard traits for structs where possible to reduce lines of code and improve readability.

### CODE REVIEW

**Comment by kennykerr** [comment]
Looks like deriving `Debug` for function pointers is new in Rust 1.70 so that'll be the new MSRV for the `windows` crate.

File: crates/libs/bindgen/src/rust/structs.rs [hunk 1]
```diff
@@ -135,110 +135,48 @@ fn gen_windows_traits(writer: &Writer, def: metadata::TypeDef, name: &TokenStrea
...
-            }
-        });
+        let features = writer.cfg_features(cfg);
 
         quote! {
             #features
             impl Clone for #name {
                 fn clone(&self) -> Self {
-                    Self { #(#fields),* }
+                    unsafe { core::mem::transmute_copy(self) }
```
Comment (dpaoliello) [inline crates/libs/bindgen/src/rust/structs.rs:179]: I'd rather if you left the previous implementation, this feels incredibly dangerous....

**Review by dpaoliello — CHANGES_REQUESTED** [review]

File: crates/libs/bindgen/src/rust/structs.rs (hunk 1 above)
Comment (kennykerr) [reply crates/libs/bindgen/src/rust/structs.rs:179]: This is the previous implementation! 🥲 I've just moved it around some, and this is the last bit that's not derived. I have some more work to do to eliminate this but yes that's the goal. This PR didn't actually add any new manual implementations. This is just the last implementation that remains, for now.

Comment (dpaoliello) [reply crates/libs/bindgen/src/rust/structs.rs:179]: Ah, I see now, GitHub's diffing was confusing me :)

**Review by dpaoliello — APPROVED** [review]

**Pull request merged**